src/ProjectTemplates/Web.ProjectTemplates/content/BlazorWeb-CSharp/BlazorWeb-CSharp/Identity/IdentityRedirectManager.cs

```diff
@@ -0,0 +1,63 @@
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.AspNetCore.Components;
+
+namespace BlazorWeb_CSharp.Identity;
+
+internal sealed class IdentityRedirectManager(
+    NavigationManager navigationManager,
+    IHttpContextAccessor httpContextAccessor)
+{
+    public const string StatusCookieName = "Identity.StatusMessage";
+
+    private static readonly CookieBuilder _statusCookieBuilder = new CookieBuilder
+    {
+        SameSite = SameSiteMode.Strict,
+        HttpOnly = true,
+        IsEssential = true,
+        MaxAge = TimeSpan.FromSeconds(5),
+    };
+
+    [DoesNotReturn]
+    public void RedirectTo(string uri)
```

Comment:
There's a lot of code that passes in root-relative URLs (e.g., `/Account/Blah`). Can you confirm those will be handled correctly even if there's a nonempty pathbase? I'm not certain that's accounted for in the logic here, but may be wrong!